src/main/java/io/odpf/firehose/sink/SinkFactory.java

```diff
@@ -67,6 +70,16 @@ public void init() {
                 logSinkFactory = new LogSinkFactory(config, statsDReporter);
                 logSinkFactory.init();
                 return;
+            case REDIS:
+                redisSinkFactory = new RedisSinkFactory(
+                        ConfigFactory.create(RedisSinkConfig.class, config),
+                        statsDReporter);
+                try {
+                    redisSinkFactory.init();
+                } catch (IOException e) {
+                    throw new IllegalArgumentException("Exception occurred while creating sink", e);
```

Comment:
ah. no. We should not catch it. 
The init() itself should throw RuntimeException.

Reply:
it throws IOException, we'll need to catch and throw IllegalArgumentException in depot then

Reply:
done